src/extension/background-script/actions/webln/enable.ts

```diff
@@ -38,17 +43,24 @@ const enable = async (message: MessageAllowanceEnable, sender: Sender) => {
           if (!allowance.id) {
             return { data: { error: "id is missing" } };
           }
-          await db.allowances.update(allowance.id, {
-            enabled: true,
-            name: message.origin.name,
-            imageURL: message.origin.icon,
-          });
+          if (allowance.enabledFor) {
+            const updatedEnabledFor = allowance.enabledFor.includes("webln")
```

Comment:
why do we do a check here when further up there is `allowance.enabledFor?.includes("webln")` further up?

Reply:
Or is the check above wrong?

Reply:
We need to make sure we can update an allowance to include multiple providers, so I think the above should be removed?

Also, maybe we could convert the array to a set and back to avoid the tertiary operator

Reply:
we don't want to duplicate the entry next time. that is why we need to check provider name is already there or not

to reproduce: unlock the extension and do the provider call. if we don't do that duplicate entries will be there in the list

Reply:
@pavanjoshi914 how does it work when you have the allowance.enabledFor = ["nostr"] and then you try to enable it to include "webln". It doesn't seem possible with the current checks?

Reply:
@pavanjoshi914 oh sorry, I read it wrong. Thanks!

This comment would be good to fix though I think: `maybe we could convert the array to a set and back to avoid the tertiary operator`

Reply:
resolved